games/rtype/src/System/EventSystem.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+** EPITECH PROJECT, 2023
+** RType
+** File description:
+** EventSystem.cpp
+*/
+
+#include "EventSystem.hpp"
+
+#include "Components.hpp"
+#include "OtterGraphic.hpp"
+
+namespace Otter::Game::RType::System::Event {
+
+    void PlayerMovementEvent(auto& velocity, int keyPressed)
+    {
+        if (keyPressed == Otter::Graphic::Raylib::RaylibKeyboard::RaylibKey::LEFT)
+            velocity->accelerationDirection.x += -1;
+        if (keyPressed == Otter::Graphic::Raylib::RaylibKeyboard::RaylibKey::RIGHT)
+            velocity->accelerationDirection.x += 1;
+        if (keyPressed == Otter::Graphic::Raylib::RaylibKeyboard::RaylibKey::UP)
+            velocity->accelerationDirection.y += -1;
+        if (keyPressed == Otter::Graphic::Raylib::RaylibKeyboard::RaylibKey::DOWN)
+            velocity->accelerationDirection.y += 1;
+    }
+
+    void PollEvent(Otter::Core::Orchestrator& ref)
+    {
+        auto const& players = ref.get_components<Otter::Games::RType::Components::Player>();
+        auto& velocities = ref.get_components<Otter::Games::RType::Components::Velocity>();
```

Comment:
Care actuali if you want to build a entity static who for exemple despawn or respawn when a key is pressed, need to have a velocity composant or it is a static object ?

Reply:
same with player actuali this system work only for player with keyboard imput.

it is not a real issue but juste feedback for advice. if you know the fact that i say don't change otherwith take note and modify to corespond what you realy wanted